build.sbt

```diff
@@ -231,7 +231,8 @@ lazy val `quill-engine` =
         "com.typesafe.scala-logging" %% "scala-logging" % "3.9.5",
         ("com.github.takayahilton"   %% "sql-formatter" % "1.2.1").cross(CrossVersion.for3Use2_13),
         "io.suzaku"                  %% "boopickle"     % "1.4.0",
-        "com.lihaoyi"                %% "pprint"        % "0.8.1"
+        "com.lihaoyi"                %% "pprint"        % "0.8.1",
+        "com.google.guava"            % "guava"         % "32.1.2-jre"
```

Comment:
I'd prefer to avoid to introduce Guava

Reply:
A concurrent cache is complex and easy to have bugs. The bugs related to concurrency is also hard to notice and fix. I don't really have the confidence to implement it from scratch perfectly.

If you'd like to import other cache implementations like zio-cache, I'm fine as well. However I think it's wise to keep quill-engine out of zio since it not only imports a library but also introduces a new coding style as well.

If the user is already dealing multiple versions of Guava, they should already have the version override in the package manager (if needed) anyway. The cache API in Guava is pretty stable so the code should work with other Guava versions.

In general I don't like adding a dependency as well. But in this case it seems worth it.

Reply:
According to Guava doc, it's preferrable to use Caffeine: https://guava.dev/releases/31.0-jre/api/docs/com/google/common/cache/CacheBuilder.html#:~:text=Caffeine's%20types%20are%20better%20than,backed%20by%20a%20Guava%20CacheLoader%20.

Can you please replace Guava with Caffeine, please?

Reply:
@guizmaii Good catch! I missed it since it's not in Guava's wiki page about cache. Pushed a new commit to use Caffeine.

Reply:
Thanks @wb14123!